src/storage/mapstorage/MapStorage.ts

```diff
@@ -0,0 +1,33 @@
+import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
+
+/**
+ * A simple storage solution that can be used for internal values that need to be stored.
+ */
+export interface MapStorage<T> {
```

Comment:
I think at this point we can establish that I'm bad at naming things. The idea was to specifically have support for storing values under ResourceIdentifiers, but there's no specific reason why this interface can't be extended generic keys as well. I did want to make sure we tested the path string of the ResourceIdentifier and not the object itself as key in case there would be equality problems there.